MarsQACompetitionTaskNUnit/Pages/EducationPage.cs

```diff
@@ -0,0 +1,326 @@
+                UniversityTextbox.SendKeys(Keys.Tab);  // Trigger blur event
+                Console.WriteLine("UniversityTextbox cleared.");
+                if (!string.IsNullOrEmpty(University))
+                {
+
+                    UniversityTextbox.SendKeys(University);
+                }
+                else
+                {
+                    Console.WriteLine("UniversityTextbox remains empty.");
+                }
+
+                CountryDropdown.Click();
+                WaitUtils.WaitToBeVisible(driver, "Xpath", "CountryDropdown", 30);
+                IWebElement EditCountryOption = driver.FindElement(By.XPath("//*[@value='" + Country + "']"));
+                EditCountryOption.Click();
+                Console.WriteLine($"CountryDropdown set to: {Country}");
+
+                TitleDropdown.Click();
+                WaitUtils.WaitToBeVisible(driver, "Xpath", "TitleDropdown", 30);
+                IWebElement TitleOption = driver.FindElement(By.XPath("//*[@value='" + Title + "']"));
+                TitleOption.Click();
+                Console.WriteLine($"TitleDropdown set to: {Title}");
+
+                // Update Degree if a value is provided
+                driver.Manage().Timeouts().ImplicitWait = TimeSpan.FromSeconds(30);
+                DegreeTextbox.Clear();
+                DegreeTextbox.SendKeys(Keys.Tab);  // Trigger blur event
+                Console.WriteLine("DegreeTextbox cleared.");
+                if (!string.IsNullOrEmpty(Degree))
+                {
+
+                    DegreeTextbox.SendKeys(Degree);
+                }
+                else
+                {
+                    Console.WriteLine("DegreeTextbox remains empty.");
+                }
+
+                GraduationYearDropdown.Click();
+                WaitUtils.WaitToBeVisible(driver, "Xpath", "GraduationYearDropdown", 30);
+                IWebElement GraduationYearOption = driver.FindElement(By.XPath("//*[@value='" + GraduationYear + "']"));
+                GraduationYearOption.Click();
+                Console.WriteLine($"GraduationYearDropdown set to: {GraduationYear}");
+
+                WaitUtils.WaitToBeClickable(driver, "Xpath", "UpdateButton", 30);
+                UpdateButton.Click();
+                Console.WriteLine("Update button clicked.");
+                Thread.Sleep(5000);
+            }
+
+            catch (Exception ex)
+            {
+                Console.WriteLine($"Exception occurred while editing education record: {ex.Message}");
+            }
+        }
+    
+
+        public void DeleteLastEducationRecords()
+        {
+
+            LastDeletePencilIcon.Click();
+            Thread.Sleep(3000);
+        }
+
+        public void ClearEducation()
+        {
+
+            try
+            {
+                int rowCount = EducationRows.Count;
+            
+                for (int i = 1; i <= rowCount; i++)
+                {
+                    DeleteLastEducationRecords();
```

Comment:
best would be again to loop through the table values

Reply:
updated loop through the table values